common/persistence/executionManager.go

```diff
@@ -642,9 +642,12 @@ func (m *executionManagerImpl) SerializeWorkflowMutation(
 	if err != nil {
 		return nil, err
 	}
-	checksumData, err := m.serializer.SerializeChecksum(input.Checksum, common.EncodingTypeJSON)
-	if err != nil {
-		return nil, err
+	var checksumData *DataBlob
+	if len(input.Checksum.Value) > 0 {
```

Comment:
`serializer.SerializerChecksum` does nil check but we are losing that with this change. Can `input.Checksum` be nil? Either way I'd move this length check to `SerializeChecksum`